### ISSUE TITLE
Is Zeno suitable for federated learning settings?

### ISSUE THREAD

**Issue by MetaT1an** [IBM/federated-learning-lib#15]
I noticed that Zeno is supported as a fusion algorithm in this lib.

The key idea of Zeno is to respectively compute the score of uploaded gradients, the score represents the degree of descendant of the loss function. Since the algorithm is performed in the server, data samples are also needed to finish score computation. 

Consequently, in the federated learning settings, are the required data samples available in the server for zeno to consume? To my understanding, the training data is distributed among different participants, only testing data is preserved in the server.

So, I am confused about the usability of zeno in FL settings.

**Comment by mathsinn** [comment]
@MetaT1an Thank you for your interest in IBM federated learning!

As you actually pointed out, Zeno works on the gradients provided by the different parties. It does not require the local training data - those will stay within the parties' premises - but only the gradients computed on those data, which are exchanged during the federated learning.

Hope this helps - please let us know if you have any additional questions.

**Comment by Nathalie-B** [comment]
One more comment, @MetaT1an  Zeno assumes that the aggregator has a _validation set_ that is _different_ to the training set. Party's data is always kept private. Thus, this method is only suitable if the aggregator has access to a validation set.

**Comment by MetaT1an** [comment]
Thank for your @Nathalie-B @mathsinn patience and kind reply. 

Since this lib is desinged mainly for enterprise environment, a validation set is  easier to be obtained than that in devices-targeted enviroment.  

Thus, Zeno works.
